### PR TITLE
FSR support.

### PR DESCRIPTION
**Pull request by DadSchoorse**
-U or Super + U to enable FSR for layer 0.

Might be better to disable fp16 until https://gitlab.freedesktop.org/mesa/mesa/-/issues/5002 is resolved.

### CODE REVIEW

**Comment by Joshua-Ashton** [comment]
Shouldn't it be possible to merge the second rcas step into our composite dispatch to eliminate the extra image, dispatch, barriers, etc there?

**Comment by DadSchoorse** [comment]
Yes that should be possible.

**Comment by DadSchoorse** [comment]
I merged rcas and the composite step now

**Comment by DadSchoorse** [comment]
There's one issue left: This increases startup times so much that it times out.

**Comment by Joshua-Ashton** [comment]
@DadSchoorse Can we do lazy pipeline compilation for FSR pipelines or something?

**Comment by DadSchoorse** [comment]
@Joshua-Ashton We now compile pipelines on a dedicated thread, with a lazy fallback if the requested pipeline wasn't compiled yet.

**Review by Joshua-Ashton — CHANGES_REQUESTED** [review]

File: src/steamcompmgr.cpp [hunk 1]
```diff
@@ -1506,6 +1506,8 @@ paint_all()
 				}
 				// Just draw focused window as normal, be it Steam or the game
 				paint_window(w, w, &composite, &pipeline, global_focus.cursor, PaintWindowFlag::BasePlane | PaintWindowFlag::DrawBorders, 1.0f);
+
+				composite.useFSRLayer0 = g_fsrUpscale;
```
[the remark this answers is not in the document]
Comment (Joshua-Ashton) [reply src/steamcompmgr.cpp]: Can also remove this in vulkan_composite.

**Comment by Bluscream** [comment]
Stop being so smart

**Pull request merged**